tools/autograd/derivatives.yaml

```diff
@@ -562,7 +562,7 @@
 
 - name: index_copy_(Tensor(a!) self, int dim, Tensor index, Tensor source) -> Tensor(a!)
   self: grad.clone().index_fill_(dim, index, 0)
-  source: grad.index_select(dim, index).expand_as(source)
+  source: "source.dim() > 0 ? grad.index_select(dim, index).expand_as(source) : grad.index_select(dim, index.squeeze(0))"
```

Comment:
Please add a comment above this gradient explaining this source formula.

@ngimel asks, "Is the proper fix for this updating index_select()? If so, let's file an issue for that."

Reply:
So, what's happening here is that, for a 0-dimensional source, `expand_as(source)` fails.  I believe that it should not fail if the input on which it is being applied is of shape `()` or `(1,)`. This might be an edge-case for  `expand_as` that might not be tested, and it might not be implemented correctly. Should I file an issue about this, or is this an expected behaviour of `expand_as`?

Reply:
Thank you for this explanation. An issue would be a great, and a comment linking the issue here would be awesome.

Reply:
Filed here: https://github.com/pytorch/pytorch/issues/53351